### ISSUE TITLE
Update mailjet template mjml version should permit to trigger rebuild html version Mailjet side

### ISSUE THREAD

**Issue by axi** [mailjet/mailjet-apiv3-php#191]
Hi, 
we are in a process of translating all our templates in different languages.

Here is the process we figured out and which would to make sense:
- Templates in french (initial language) were/are initialy created with Passport
- We duplicate each template and template detail (through api), adding a suffix _en to it's name
- We get the template detail through API, especialy MJMLContent (a json object)
- Translation is made on needed contents
- We upload update MJMLContent

From here what I would like:
- A way to trigger the rebuild of html version Mailjet side, same way it's done in Passport

What I have to do currently (with all the problems it triggers):
- convert localy MJML (json version) to MJML (xml version) as mjml tool doesn't compile properly to html (empty <body></body>)
- compile it to html
- update both version (MJML & html) on template detail

 With the current solution, parts of MJML are not compiled in the same way that Passport do. For exemple, compiling that localy result in the section existing in html version although it doesn't look supposed to be (passport.hidden = true):
```
{
    "tagName": "mj-section",
    "attributes": {
        "padding": "20px 0px 20px 0px",
        "background-repeat": "repeat",
        "background-size": "auto",
        "text-align": "center",
        "vertical-align": "top",
        "passport": {
            "hidden": true,
            "header": true,
            "id": "pkcfpzoekcfpzek"
        }
    },
    "children": ["..."]
}
```

A backup solution (not tried yet) would be to upload MJMLContent only then go in Passport, open the template and trigger the rebuild...

To me, a simple "recompileHtml: bool" param on https://dev.mailjet.com/email/reference/templates/#v3_post_template_template_ID_detailcontent update would be perfect.
Is it something you're working on ?
Or is the workflow I created wrong in a way ?

**Comment by axi** [comment]
#79 & #76 are related

**Comment by axi** [comment]
> A backup solution (not tried yet) would be to upload MJMLContent only then go in Passport, open the template and trigger the rebuild...
Tried it, Passport doesn't care about what's updated through api, I never see my modified content in Passport.
Looks like there is no way to update a Passport created template though api and hope to be able to use Passport to modify it afterwards

**Comment by ngarnier** [comment]
Hi, thanks for this feedback. 2 things here: 

- Indeed, the only way to render MJML to HTML in Mailjet is through Passport. We have no plan to change this at this time
- Passport does care about the changes you make to a template via API, **but** when you update a template via API, you're updating the publish version, whereas when you open a template in Passport, you're editing the draft version. If you look at the "Previously published version", you'll see the content you sent via API as the latest published version

I'll close this ticket as it's not related to the PHP wrapper, but feel free to open a support ticket to submit your feature request.

**Comment by axi** [comment]
thanks @ngarnier. I had a look on this latest published version but no changes appeared. Will try another time

**Comment by ngarnier** [comment]
The preview of the latest published version is the HTML, so if you only updated the MJML you won't see any difference. You have to restore that latest published version so that it's opened in the builder and the MJML is rendered again.

**Comment by axi** [comment]
I'll try that tomorrow morning thanks

**Comment by axi** [comment]
Tried again:
- uploaded the new MJML (and updated the subject header).
- in passport, the draft version date has been updated to the date I used PUT templateDetail. Looks encouraging.
- I click on the the latest published version, then on restore button, hit the confirm button within the modale
- Passport opens on the published version but my changes are not there.
- If I use api GET on templateDetail, my modification are there
- If I send a mail using this template, I receive a **not** modified version (but received mail subject is correct (meaning the updated one)...
